File: Pbm/Codeforces/2026.1-2026.3/2199C.cpp
//2026/3/21
#include <bits/stdc++.h>
using namespace std;
#define ll long long
#define endl '\n'

void solve() {
  int k; cin >> k;
  if(k == 1) {
    cout << "YES" << endl << 1 << endl << '*' << endl << '.' << endl;
    return;
  } 
  string s1, s2;
  bool flag = 0;
  if((k-3)%5 == 0 || (k-6)%5 == 0) {
    k -= 3;
    s1 += "*.";
    flag = 1;
  } 
  if(k%5 == 0 || (k-3)%5 == 0) {
    int num = k/5;
    k -= num*5;
    for(int i = 1; i <= num; ++i) s1 += ".*."; 
    flag = 1;
  } 
  if(k == 3) s1 += ".*";
  if(flag) {
    cout << "YES" << endl << s1.length() << endl << s1 << endl;
    for(int i = s1.length(); i >= 1; --i) cout << '.';
    cout << endl;
  } else cout << "NO" << endl;
} 

int main() {
  cin.tie(0) -> sync_with_stdio(0);
  int T = 1; cin >> T;
  while(T--) solve();
  return 0;
}

File: Pbm/Codeforces/2026.1-2026.3/2199C.kt
import java.io.PrintWriter
import java.util.StringTokenizer

fun solve(out: PrintWriter) {
    val k = readLine()!!.toInt()

    if (k == 1) {
        out.println("YES")
        out.println(1)
        out.println("*")
        out.println(".")
        return
    }

    val s1Builder = StringBuilder()
    var remainingK = k
    var flag = false

    // Condition 1: Check if (k-3) or (k-6) is divisible by 5
    if ((remainingK - 3) % 5 == 0 || (remainingK - 6) % 5 == 0) {
        remainingK -= 3
        s1Builder.append("*.")
        flag = true
    }

    // Condition 2: Check if k or (k-3) is divisible by 5
    if (remainingK % 5 == 0 || (remainingK - 3) % 5 == 0) {
        val num = remainingK / 5
        remainingK -= num * 5
        repeat(num) {
            s1Builder.append(".*.")
        }
        flag = true
    }

    // Final addition
    if (remainingK == 3) {
        s1Builder.append(".*")
    }

    val s1 = s1Builder.toString()

    if (flag) {
        out.println("YES")
        out.println(s1.length)
        out.println(s1)
        repeat(s1.length) {
            out.print('.')
        }
        out.println()
    } else {
        out.println("NO")
    }
}

fun main() {
    val out = PrintWriter(System.out.bufferedWriter(), false)
    val T = readLine()!!.toInt()
    repeat(T) {
        solve(out)
    }
    out.flush()
}